fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/DimensionRow.java

```diff
@@ -63,4 +77,18 @@ public int compareTo(DimensionRow that) {
         return entrySet().stream()
                 .collect(StreamUtils.toLinkedMap(entry -> entry.getKey().getName(), Map.Entry::getValue));
     }
+
+    /**
+     * Create a copy of a dimension row, and then transform it using a mapping function.
```

Comment:
It isn't really clear from the documentation that when we mean "transform" it we mean, apply a mapping to each dimension field's value. Something like:

```
Copies a DimensionRow and transforms its fields using the specified mapper.

@param row The row to be transformed
@param mapper  A function that takes a DimensionField and its current value, and returns the field's new value.

@return A copy of the DimensionRow with its fields transformed by the specified function
```

Reply:
indeed, clearer.